### ISSUE TITLE
Replace kubectl by direct API calls through curl/jq and python-kubernetes

### ISSUE THREAD

**Issue by timn** [ceph/ceph-container#996]
Inside the container it would make sense to use curl and [jq](https://stedolan.github.io/jq/) rather than kubectl. For an example have a look at a [wait-services script](https://github.com/timn/docker-robotics/blob/master/rcll-sim/scripts/wait-services) we use, relying on exactly those two.

Benefits:
- drop rather heavy-weight kubectl (binary is about 170 MB)
- tools are light-weight and may be included already anyway
- jq does all the parsing and gives precisely the information queried, you can also easily run mutliple queries (but still fetch only once)

For the integrated python code, such as check_zombie_mons.py, the Kubernetes Python client should be used. An example how this looks is in https://github.com/timn/rcll-sim-cluster/blob/master/sim-ctrl/pod_controller.py.  It adds in about 20 MB (still much less than kubectl).

Something similar will probably hold for etcd (etcdctl also is about 20MB), but I have not used it as much directly to be sure.

Extracted from a discussion in #994, @ErwanAliasr1 @Raboo

**Comment by ErwanAliasr1** [comment]
Thanks @timn for opening this issue. Added the label on it.

**Comment by timn** [comment]
For K8s, which I use, there is a number of "get pods", and one "create configmap" and one "delete" for an unresponsive mon. These are the calls that I proposed to replace in the first place. If you then look at config.kv.etcd, that makes many more calls, especially setting various bits of information. From this perspective, a unified interface does not look very attractive, especially since this already is mostly isolated and k8s is mostly read-only, while etcd is heavily read-write (the one create for k8s does not seem to translate to anything in the etcd case, and mon monitoring also seems to be implemented for k8s only).

So, as a goal for this ticket I'd simply work on removing kubectl. Considering the simplicity of check_zombie_mons, this can probably be translated to a bash script using curl/jq and therefore python-kubernetes would not be necessary. I would see a more general take to unify backends as another issue that requires some time coming up with a solid design document and test cases.
